src/api/uniqueSdk/collectionController.ts

```diff
@@ -0,0 +1,119 @@
+import { ICollectionController, TransactionOptions } from '../chainApi/types';
+import { NFTCollection, NFTToken, TokenId } from '../chainApi/unique/types';
+import { Sdk } from '@unique-nft/sdk';
+import { Settings } from '../restApi/settings/types';
+import { getTokenImage } from './utils/imageUtils';
+import config from '../../config';
+
+const { IPFSGateway } = config;
+
+export class UniqueSDKCollectionController implements ICollectionController<NFTCollection, NFTToken> {
+  private sdk: Sdk;
+  private settings;
+  constructor(sdk: Sdk, settings: Settings) {
+    this.sdk = sdk;
+    this.settings = settings;
+  }
+
+  async confirmSponsorship(collectionId: number, options: TransactionOptions): Promise<void> {
```

Comment:
signPayloadJson is not optional for this function, but it is optional for "TransactionOptions". We better put it as param outside of "options", since it's not an option but a mandatory param